tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CodeownersTools/CodeownersTools.cs

```diff
@@ -0,0 +1,443 @@
+                    var serviceOwnersValue = commandParser.GetValueForOption(serviceOwnersOption);
+                    var sourceOwnersValue = commandParser.GetValueForOption(sourceOwnersOption);
+                    var isAddingValue = commandParser.GetValueForOption(isAddingOption);
+                    var workingBranchValue = commandParser.GetValueForOption(workingBranchOption);
+
+                    var addResult = await UpdateCodeowners(
+                        repoValue ?? "",
+                        typespecPathValue ?? "",
+                        pathValue ?? "",
+                        serviceLabelValue ?? "",
+                        serviceOwnersValue?.ToList() ?? new List<string>(),
+                        sourceOwnersValue?.ToList() ?? new List<string>(),
+                        isAddingValue,
+                        workingBranchValue ?? "");
+                    output.Output(addResult);
+                    return;
+                case validateCodeownersEntryCommandName:
+                    var validateRepo = commandParser.GetValueForOption(repoOption);
+                    var validateServiceLabel = commandParser.GetValueForOption(serviceLabelOption);
+                    var validateRepoPath = commandParser.GetValueForOption(pathOptionOptional);
+
+                    var validateResult = await ValidateCodeownersEntryForService(
+                        validateRepo ?? "",
+                        validateServiceLabel,
+                        validateRepoPath);
+                    output.Output(validateResult);
+                    return;
+                default:
+                    SetFailure();
+                    output.OutputError($"Unknown command: '{command}'");
+                    return;
+            }
+        }
+
+        [McpServerTool(Name = "UpdateCodeowners"), Description("Adds or deletes codeowners for a given service label or path in a repo.")]
+        public async Task<string> UpdateCodeowners(
+            string repo,
+            string typeSpecProjectRoot,
+            string path = null,
+            string serviceLabel = null,
+            List<string> serviceOwners = null,
+            List<string> sourceOwners = null,
+            bool isAdding = false,
+            string workingBranch = null)
+        {
+            try
+            {
+                // Validate atleast Service Label or Path.
+                if (string.IsNullOrWhiteSpace(serviceLabel) && string.IsNullOrWhiteSpace(path))
+                {
+                    throw new Exception($"Service label: {serviceLabel} and Path: {path} are both invalid. Atleast one must be valid");
+                }
+                // Check if it's management plane.
+                var isMgmtPlane = typespecHelper.IsTypeSpecProjectForMgmtPlane(typeSpecProjectRoot);
+
+                // Normalize service path
+                path = codeownersHelper.NormalizePath(path);
```

Comment:
Use a new variable like `normalizedPath`. 

This mutates the state of the `path` variable in the scope of this function. Generally, we avoid mutating variables (especially function inputs) as this makes the code more difficult to reason about.